syntax = "proto3";
import "google/protobuf/empty.proto";

// The greeting service definition.
service MeasurementProducer {
  // Sends a greeting
  rpc ProcessMeasurement (measurement_group) returns (google.protobuf.Empty) {}
}

// The greeting service definition.
service TrackProducer {
  // Sends a greeting
  rpc ProcessTrack (track_group) returns (google.protobuf.Empty) {}
}
message measurement_group{
  repeated bytes measurements = 1;
}

// The request message containing the user's name.
message measurement {
  double  x = 1;
  double y = 2;
  double z = 3;
  double time = 4;
}

message track_group{
  repeated bytes tracks = 1;
}

// The response message containing the greetings
message track {
  repeated bytes measurements = 1;
  double x_velocity = 2;
  double y_velocity = 3;
  double z_velocity = 4;
}